WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayEducationView.swift

```diff
@@ -67,3 +67,24 @@ struct TapToPayEducationView: View {
         })
     }
 }
+
+// MARK: - Hosting Controller
+
+final class TapToPayEducationViewHostingController: UIHostingController<TapToPayEducationView> {
+    init(onDismiss: @escaping () -> Void) {
+        let viewModel = TapToPayEducationViewModel(flow: .onboarding, onDismiss: onDismiss)
+        super.init(rootView: TapToPayEducationView(viewModel: viewModel))
+
+        viewModel.onDismiss = { [weak self] in
+            guard let self else { return }
+
+            dismiss(animated: true) {
+                onDismiss()
+            }
+        }
```

Comment:
This whole dance around self is really annoying 😢 

Is it still actually required? Do hosting controllers still not respect the dismiss Environment value? [This post suggests it _should_ work, since you do have a NavigationStack...](https://www.devgem.io/posts/dismissing-a-swiftui-view-presented-via-uihostingcontroller) (I wouldn't bother with the more complex solutions there though...)

Reply:
I did this annoying dance since I needed to know when the education view 'did' dismiss rather than 'will' dismiss. I'll check the post you linked, thanks!